mantidproject/mantid
# IndexProperty Implementation Prototype

**DO NOT MERGE**
**NOT READY FOR REVIEW**
This is an alternative implementation to the implementation of an IndexProperty see [here](https://github.com/mantidproject/mantid/pull/19699) for the first attempt. This [design](https://github.com/mantidproject/documents/pull/42) outlines the rational behind developing this property type. 

**To test:**
Code review. Ensure the property displays correctly in the GUI and can be accessed from python correctly. The `ChangePulsetime2` algorithm was implemented specifically to demonstrate the use of this property.

Fixes #19130. 

**Release Notes** 
Release notes need to be updated in due course and the concepts pages.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
